File: repos_estudiantes_ALVAREZ_tercera_entrega/AJATA_CANAVIRI_NICOL_MAYRA/Persistencia_de_Objetos/Ejercicio1/Java/Main.java
package Persistencia_de_Objetos.Ejercicio1.Java;
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;

public class Main {
    public static void main(String[] args) {
        String nombreArchivo = "empleados_ejercicio1.txt";
        ArchivoEmpleado archivoEmpleado = new ArchivoEmpleado(nombreArchivo);

        archivoEmpleado.limpiarArchivo();

        System.out.println("\n--- Guardando Empleados ---");
        Empleado emp1 = new Empleado("Maria Celeste", 30, 55000.00f);
        Empleado emp2 = new Empleado("Carlos Johnson", 25, 48000.50f);
        Empleado emp3 = new Empleado("Abel Fernandez", 45, 72000.75f);
        Empleado emp4 = new Empleado("Santiago Barrancos", 32, 60000.00f);

        archivoEmpleado.guardarEmpleado(emp1);
        archivoEmpleado.guardarEmpleado(emp2);
        archivoEmpleado.guardarEmpleado(emp3);
        archivoEmpleado.guardarEmpleado(emp4);

        System.out.println("\n--- Buscando Empleados por Nombre ---");
        // b) 
        String nombreBusqueda1 = "Carlos Johnson";
        Empleado empleadoEncontrado1 = archivoEmpleado.buscaEmpleado(nombreBusqueda1);
        if (empleadoEncontrado1 != null) {
            System.out.println("Empleado '" + nombreBusqueda1 + "' encontrado: " + empleadoEncontrado1);
        } else {
            System.out.println("Empleado '" + nombreBusqueda1 + "' no encontrado.");
        }

        String nombreBusqueda2 = "Maria Celeste";
        Empleado empleadoEncontrado2 = archivoEmpleado.buscaEmpleado(nombreBusqueda2);
        if (empleadoEncontrado2 != null) {
            System.out.println("Empleado '" + nombreBusqueda2 + "' encontrado: " + empleadoEncontrado2);
        } else {
            System.out.println("Empleado '" + nombreBusqueda2 + "' no encontrado.");
        }

        String nombreBusqueda3 = "David Lopez";
        Empleado empleadoEncontrado3 = archivoEmpleado.buscaEmpleado(nombreBusqueda3);
        if (empleadoEncontrado3 != null) {
            System.out.println("Empleado '" + nombreBusqueda3 + "' encontrado: " + empleadoEncontrado3);
        } else {
            System.out.println("Empleado '" + nombreBusqueda3 + "' no encontrado.");
        }

        System.out.println("\n--- Buscando Empleado con Salario Mayor ---");
        // c) 
        float sueldoMinimo1 = 50000.00f;
        Empleado empleadoMayorSalario1 = archivoEmpleado.mayorSalario(sueldoMinimo1);
        if (empleadoMayorSalario1 != null) {
            System.out.println("Primer empleado con salario mayor a " + String.format("%.2f", sueldoMinimo1) + ": " + empleadoMayorSalario1);
        } else {
            System.out.println("Ningun empleado encontrado con salario mayor a " + String.format("%.2f", sueldoMinimo1) );
        }

        float sueldoMinimo2 = 75000.00f;
        Empleado empleadoMayorSalario2 = archivoEmpleado.mayorSalario(sueldoMinimo2);
        if (empleadoMayorSalario2 != null) {
            System.out.println("Primer empleado con salario mayor a " + String.format("%.2f", sueldoMinimo2) + ": " + empleadoMayorSalario2);
        } else {
            System.out.println("Ningun empleado encontrado con salario mayor a " + String.format("%.2f", sueldoMinimo2));
        }

        System.out.println("\nContenido final del archivo '" + nombreArchivo + "':");
        try (BufferedReader reader = new BufferedReader(new FileReader(nombreArchivo))) {
            String line;
            while ((line = reader.readLine()) != null) {
                System.out.println(line);
            }
        } catch (IOException e) {
            System.err.println("Error al leer el archivo para mostrar su contenido final: " + e.getMessage());
        }
    }
}

File: repos_estudiantes_ALVAREZ_tercera_entrega/AJATA_CANAVIRI_NICOL_MAYRA/Persistencia_de_Objetos/Ejercicio1/Python/main.py
from Empleado import Empleado
from ArchivoEmpleado import ArchivoEmpleado

if __name__ == "__main__":
    nombre_archivo = "empleados_ejercicio1.txt"
    archivo_empleado = ArchivoEmpleado(nombre_archivo)

    archivo_empleado.limpiar_archivo()

    print("\n--- Guardando Empleados ---")
    emp1 = Empleado("Maria Celeste", 30, 55000.00)
    emp2 = Empleado("Carlos Johnson", 25, 48000.50)
    emp3 = Empleado("Abel Fernandez", 45, 72000.75)
    emp4 = Empleado("Santiago Barrancos", 32, 60000.00) 

    archivo_empleado.guardar_empleado(emp1)
    archivo_empleado.guardar_empleado(emp2)
    archivo_empleado.guardar_empleado(emp3)
    archivo_empleado.guardar_empleado(emp4)

    print("\n--- Buscando Empleados por Nombre ---")
    # b) 
    nombre_busqueda1 = "Carlos Johnson"
    empleado_encontrado1 = archivo_empleado.busca_empleado(nombre_busqueda1)
    if empleado_encontrado1:
        print(f"Empleado '{nombre_busqueda1}' encontrado: {empleado_encontrado1}")
    else:
        print(f"Empleado '{nombre_busqueda1}' no encontrado.")

    nombre_busqueda2 = "Maria Celeste"
    empleado_encontrado2 = archivo_empleado.busca_empleado(nombre_busqueda2)
    if empleado_encontrado2:
        print(f"Empleado '{nombre_busqueda2}' encontrado: {empleado_encontrado2}")
    else:
        print(f"Empleado '{nombre_busqueda2}' no encontrado.")

    nombre_busqueda3 = "David Lopez"
    empleado_encontrado3 = archivo_empleado.busca_empleado(nombre_busqueda3)
    if empleado_encontrado3:
        print(f"Empleado '{nombre_busqueda3}' encontrado: {empleado_encontrado3}")
    else:
        print(f"Empleado '{nombre_busqueda3}' no encontrado.")

    print("\n--- Buscando Empleado con Salario Mayor ---")
    # c)
    sueldo_minimo_1 = 50000.00
    empleado_mayor_salario1 = archivo_empleado.mayor_salario(sueldo_minimo_1)
    if empleado_mayor_salario1:
        print(f"Primer empleado con salario mayor a {sueldo_minimo_1:.2f}: {empleado_mayor_salario1}")
    else:
        print(f"Ningún empleado encontrado con salario mayor a {sueldo_minimo_1:.2f}.")

    sueldo_minimo_2 = 75000.00
    empleado_mayor_salario2 = archivo_empleado.mayor_salario(sueldo_minimo_2)
    if empleado_mayor_salario2:
        print(f"Primer empleado con salario mayor a {sueldo_minimo_2:.2f}: {empleado_mayor_salario2}")
    else:
        print(f"Ningun empleado encontrado con salario mayor a {sueldo_minimo_2:.2f}.")

    
    print(f"\nContenido final del archivo '{nombre_archivo}':")
    try:
        with open(nombre_archivo, 'r') as f:
            print(f.read())
    except FileNotFoundError:
        print("El archivo no existe aun")
